filters/diag/diag.go

```diff
@@ -417,3 +451,89 @@ func (t *throttle) Response(ctx filters.FilterContext) {
 	rsp := ctx.Response()
 	rsp.Body = t.goThrottle(rsp.Body, true)
 }
+
+func (j *jitter) Name() string {
+	switch j.typ {
+	case normalDist:
+		return "jitterNormal"
+	case uniformDist:
+		return "jitterUniform"
+	}
+	return "unknown"
+}
+
+func (j *jitter) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var (
+		delay     time.Duration
+		interval  time.Duration
+		mu, sigma float64
+		ok        bool
+		err       error
+	)
+
+	if !(len(args) == 2 || len(args) == 4) {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if delay, err = parseDuration(args[0]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration delay %v: %w", args[0], err)
+	}
+
+	if interval, err = parseDuration(args[1]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration interval %v: %w", args[1], err)
+	}
+
+	switch j.typ {
+	case normalDist:
+		switch len(args) {
+		case 2:
+			mu = 0
+			sigma = 1
+		case 4:
+
+			mu, ok = args[2].(float64)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert mu to float64 %v", args[2])
+			}
+			sigma, ok = args[3].(float64)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert sigma to float64 %v", args[3])
+			}
+		}
+	case uniformDist:
+		if len(args) != 2 {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	default:
+		return nil, errors.New("distribution does not exist")
+	}
+
+	return &jitter{
+		typ:      j.typ,
+		delay:    delay,
+		interval: interval,
+		mu:       mu,
+		sigma:    sigma,
+	}, nil
+}
+
+func (j *jitter) Request(filters.FilterContext) {
+	d := j.delay
+
+	switch j.typ {
+	case uniformDist:
+		/* #nosec */
+		r := rand.Float64() - 0.5 // +/- \epsilon
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
```

Comment:
for delay=100ms and interval=20ms
this would sleep 100ms +/- 10ms while docs say
> The uniformJitter filter creates a jitter latency in a uniform
distribution for requests. The first parameter is the mean and the
second is the window. In the example we would sleep for 100ms+/-20ms.

so either `interval`  should be doubled here or docs fixed.

Reply:
docs fixed  thanks :)